main.go

```diff
@@ -70,6 +70,9 @@ func NewVideoStream(url string, duration time.Duration, outfile string, username
 // VideoStream.
 func (vs *VideoStream) Close() error {
 	err := vs.f.Close()
+	if err != nil {
```

Comment:
The problem is that if an error occurs in closing `f`, `res.Body` will not be closed. It would be better to attempt to close both, store the results of the attempted close in a `[]error`, then check the errors at the end of the func.